cpp/farm_ng/core/misc/conversions.h

```diff
@@ -0,0 +1,128 @@
+//    Copyright 2022, farm-ng inc.
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#pragma once
+
+#include <Eigen/Dense>
+#include <farm_ng/core/logging/logger.h>
+#include <sophus/concepts/point.h>
+#include <sophus/lie/se3.h>
+
+namespace farm_ng {
+
+/// Converts meters [m] to inches [in]
+template <class T>
+T inchesFromMeters(T m) {
+  static_assert(!std::is_integral_v<T>, "Use a floating point type.");
+  return m / 0.0254f;
+}
+
+/// Converts inches [in] to meters [m]
+template <class T>
+T metersFromInches(T inches) {
+  static_assert(!std::is_integral_v<T>, "Use a floating point type.");
+  return 0.0254 * inches;
+}
+
+/// Converts feet [ft] to meters [m]
+template <class T>
+T metersFromFeet(T feet) {
+  static_assert(!std::is_integral_v<T>, "Use a floating point type.");
+  return metersFromInches(12.0 * feet);
+}
+
+/// Converts dots to meters [m]. Assumes 72 DPI.
+template <class T>
+T metersFromDots(T dot) {
+  return metersFromInches(dot / 72.f);
+}
+
+/// Converts meters [m] to dots. Assumes 72 DPI.
+template <class T>
+T dotsFromMeters(T meters) {
+  return inchesFromMeters(meters) * 72.f;
+}
+
+/// Converts 2d vector from meters [m] to dots. Assumes 72 DPI.
+template <class T>
+Eigen::Vector2<T> dotsFromMeters(Eigen::Vector2<T> const& meters) {
+  return Eigen::Vector2<T>(
+      dotsFromMeters(meters.x()), dotsFromMeters(meters.y()));
+}
+
+/// Converts radians to degrees
+template <class T>
+T degFromRad(T rad) {
+  static_assert(!std::is_integral_v<T>, "Use a floating point type.");
+  return T(180.0) / M_PI * rad;
+}
+
+/// Converts radians to arc minutes (60th of a degree)
+template <class T>
+T arcMinuteFromRad(T rad) {
+  static_assert(!std::is_integral_v<T>, "Use a floating point type.");
+  return T(60.) * degFromRad(rad);
+}
+
+/// Converts degrees to radians
+template <sophus::concepts::RealScalarType T>
+T radFromDeg(T deg) {
+  return M_PI / T(180.0) * deg;
+}
+
+template <sophus::concepts::RealEigenDenseType TPoint>
+auto radFromDeg(TPoint const& v_deg) {
+  return radFromDeg(typename TPoint::Scalar(1.0)) * v_deg;
+}
+
+/// GeoSpherical coordinates (latitude & longitude)
+struct GeoSpherical {
```

Comment:
Not sure if this is a type we want to continue supporting for geospatial code. I could imagine it being renamed to SphericalCoord, with phi/theta instead of lat/lon.

Reply:
removed